libdnf/common/exception.cpp

```diff
@@ -59,8 +59,13 @@ std::string SystemError::get_error_message() const {
 }
 
 
-std::string format(const std::exception & e, std::size_t level) {
-    std::string ret(std::string(level, ' ') + e.what() + '\n');
+static std::string format(const std::exception & e, std::size_t level) {
+    std::string ret;
+    if (auto ex = dynamic_cast<const Error *>(&e)) {
+        ret = fmt::format("{}::{}: {}\n", ex->get_domain_name(), ex->get_name(), ex->what());
```

Comment:
This might be a topic for a separate debate (and I won't hold the PR on this), but I really don't want to print the namespaces to the end user. It's just gibberish to them to see e.g.:
```
libdnf::config::ConfigParser::InvalidValueError: Invalid value "foo" for config option "[main]::gpgcheck".
```
(the error is made up, I didn't check the actual class)

Half of that line above is noise for a normal user, and out of context for the CLI tool. It's only the second half that bears the information the user needs. I think we should simplify the messages somehow, and the first thing that comes to mind is dropping the namespace (and have a class name and message that are good enough). Ideally it wouldn't even contain the class name:
```
Error: Invalid value "foo" for config option "[main]::gpgcheck".
```
This is all the end user needs. I admit the class name is useful for us, so maybe it's a good compromise.

Another idea is to have another attribute of exceptions along the lines of:
```c++
virtual const char * get_short_id() const noexcept { return _("Configuration Error"); }
```

And then print:
```
Configuration Error: Invalid value "foo" for config option "[main]::gpgcheck".
```

That would be the nicest output and would even be translated.